autotest/autotest_services/base.py

```diff
@@ -1138,3 +1138,93 @@ def setUp(self):
                           "rasdaman tests.")
 
         super(RasdamanTestCaseMixIn, self).setUp()
+
+class WPS10XMLComparison(XMLTestCase):
+    def getFileExtension(self, part=None):
+        if part == "xml":
+            return "xml"
+        elif part == "raster":
+            return "tif"    
+
+    @staticmethod
+    def parseFileName(src) : 
+        try : 
+            with file( src ) as fid : 
+                return fid.read() 
+        except Exception as e : 
+            raise XMLParseError , "Failed to parse the \"%s\" file! %s" % ( src , str(e) ) 
+
+    def parse( self, src) :
+        return  self.parseFileName(src) 
+
+
+    def testXMLComparison(self):
+        
+        expected_path= os.path.join(
+            self.getExpectedFileDir(), self.getExpectedFileName('xml')
+        )
+
+        expectedString= self.parse(expected_path)
+        doc = etree.fromstring(expectedString)
+        encodedText= ' '.join(e.text for e in doc.xpath('//wps:ComplexData', namespaces= {'wps': 'http://www.opengis.net/wps/1.0.0'}))
+       
+        # replace the encoded data so it compare other nodes in the xml files
+        tree = etree.fromstring(self.prepareXMLData(self.getXMLData()))
+        for node in tree.xpath('//wps:ComplexData', namespaces= {'wps': 'http://www.opengis.net/wps/1.0.0'}):
+            node.text = encodedText
+            
```

Comment:
I think the intention is to copy the `wps:ComplexData` elements from the expected document to the response document, so that potential differences do not cause errors. I think that is a fair approach in principle, but I think the execution will likely not work.

I think it is best to pairwise iterate over all `ComplexData` elements, something like this:

```
expected_elems = expected_doc.xpath('//wps:ComplexData', namespaces={'wps': 'http://www.opengis.net/wps/1.0.0'})
response_elems = response_doc.xpath('//wps:ComplexData', namespaces= {'wps': 'http://www.opengis.net/wps/1.0.0'})

for expected_elem, response_elem in zip(expected_elems, response_elems):
    parent = response_elem.getparent()
    # override the response elem with the expected elem
    parent[parent.index(response_elem)] = expected_elem
```